hpx/lcos/detail/future_data.hpp

```diff
@@ -219,6 +236,47 @@ namespace detail
             return data_;
         }
 
+        // deferred execution of a given continuation
+        void run_on_completed(completed_callback_type const& on_completed)
```

Comment:
Probably keeping constness/non-constness of the passed function. Other than that probably non.